.github/workflows/aws_demo_runner.yml

```diff
@@ -0,0 +1,161 @@
+name: Localhost Server Demo Runner
+
+on:
+  push:
+    branches: ["**"]
+  pull_request:
+    branches: [main]
+  workflow_dispatch:
+
+jobs:
+  build-and-run-localhost-demos:
+    name: Build and Run Localhost Server Demos
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Clone This Repo
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      - name: Install Python3
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.8'
+
+      - name: Install libmosquitto
+        run: sudo apt-get install -y libmosquitto-dev
+
+      - name: Install CMake
+        run: |
+          curl https://cmake.org/files/v3.2/cmake-3.2.0-Linux-x86_64.tar.gz -o cmake.tar.gz
+          tar -xf cmake.tar.gz
+
+      - name: Generate SSL credentials
+        id: generate-credentials
+        uses: jasonpcarroll/CI-CD-GitHub-Actions/ssl-credential-creator@main
+
+      - name: Start localhost MQTT broker
+        id: mqtt-broker
+        uses: jasonpcarroll/CI-CD-GitHub-Actions/localhost-mqtt-broker@main
+        with:
+          root-ca-cert-path: ${{ steps.generate-credentials.outputs.root-ca-cert-path }}
+          server-priv-key-path: ${{ steps.generate-credentials.outputs.server-priv-key-path }}
+          server-cert-path: ${{ steps.generate-credentials.outputs.server-cert-path }}
+
+      - name: Start localhost HTTP server
+        id: http-server
+        uses: Eun/http-server-action@v1.0.6
+
+      - name: Configure CMake Build for Localhost servers
+        run: |
+          mkdir build && cd build
+          ../cmake-3.2.0-Linux-x86_64/bin/cmake .. \
```

Comment:
Is there a reason that the specific version of cmake needs to be specified like this? Seems like this is something that should just be able to be called usign just "cmake" from the path

Reply:
Updated to use an action from the github marketplace